_shared/project/tests/pytest_plugins/cookiecutter.py

```diff
@@ -0,0 +1,48 @@
+{% if cookiecutter.get("postgres") == "yes" and cookiecutter.get("_directory") in ["pyapp", "pyramid-app"]  %}
+from os import environ
+
+{% endif %}
+import factory.random
+import pytest
+{% if cookiecutter.get("postgres") == "yes" and cookiecutter.get("_directory") in ["pyapp", "pyramid-app"]  %}
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from {{ cookiecutter.package_name }}.db import Base
+{% endif %}
+
+
+{% if cookiecutter.get("postgres") == "yes" and cookiecutter.get("_directory") in ["pyapp", "pyramid-app"]  %}
+@pytest.fixture(scope="session")
+def db_engine():
+    engine = create_engine(environ["DATABASE_URL"])
+
+    Base.metadata.drop_all(engine)
+    Base.metadata.create_all(engine)
+
+    return engine
+
+
+@pytest.fixture(scope="session")
+def db_sessionfactory():
+    return sessionmaker()
+
+
+{% endif %}
+@pytest.fixture(scope="session", autouse=True)
+def factory_boy_random_seed():
+    # Set factory_boy's random seed so that it produces the same random values
+    # in each run of the tests.
+    # See: https://factoryboy.readthedocs.io/en/latest/index.html#reproducible-random-values
+    factory.random.reseed_random("{{ cookiecutter.github_owner }}/{{ cookiecutter.slug }}")
+{% if cookiecutter.get("_directory") == "pyramid-app" %}
+
+
+@pytest.fixture
+def pyramid_settings():
+{% if cookiecutter.get("postgres") == "yes" %}
+    return {"database_url": environ["DATABASE_URL"]}
+{% else %}
+    return {}
+{% endif %}
+{% endif %}
```

Comment:
So the cookiecutter provides a base `pyramid_settings()` fixture that might just be an empty dict or it might have the `"database_url"` setting in it. Projects can override this fixture in `conftest.py` and add in their own per-project settings.